src/pynwb/tests/test_example_usage_all.py

```diff
@@ -0,0 +1,242 @@
+    ContactsTable,
+    ExtracellularSeries,
+    Probe,
+    ProbeInsertion,
+    ProbeModel,
+)
+
+from pynwb import NWBHDF5IO, NWBFile
+
+
+def test_all_classes():
+
+    # initialize an NWBFile object
+    nwbfile = NWBFile(
+        session_description="A description of my session",
+        identifier=str(uuid.uuid4()),
+        session_start_time=datetime.datetime.now(datetime.timezone.utc),
+    )
+
+    contacts_table = ContactsTable(
+        description="Test contacts table",
+    )
+    # for demonstration, mix and match different shapes. np.nan means the radius/width/height does not apply
+    contacts_table.add_row(
+        relative_position_in_mm=[10.0, 10.0],
+        contact_id="C1",
+        device_channel=1,
+        shank_id="shank0",
+        plane_axes=[[1.0, 0.0], [0.0, 1.0]],
+        shape="circle",
+        radius_in_um=10.0,
+        width_in_um=np.nan,
+        height_in_um=np.nan,
+    )
+    contacts_table.add_row(
+        relative_position_in_mm=[20.0, 10.0],
+        contact_id="C2",
+        device_channel=2,
+        shank_id="shank0",
+        plane_axes=[[1 / np.sqrt(2), 1 / np.sqrt(2)], [-1 / np.sqrt(2), 1 / np.sqrt(2)]],
+        shape="square",
+        radius_in_um=np.nan,
+        width_in_um=10.0,
+        height_in_um=10.0,
+    )
+
+    # add the object into nwbfile.acquisition for testing
+    # TODO after integration, put this into /general/extracellular_ephys
+    nwbfile.add_acquisition(contacts_table)
+
+    pm = ProbeModel(
+        name="Neuropixels 1.0",
+        description="A neuropixels probe",
+        model="Neuropixels 1.0",
+        manufacturer="IMEC",
+        planar_contour_in_um=[[-10.0, -10.0], [10.0, -10.0], [10.0, 10.0], [-10.0, 10.0]],
+        contacts_table=contacts_table,
+    )
+    # TODO put this into /general/device_models
+    nwbfile.add_device(pm)
+
+    probe = Probe(
+        name="Neuropixels Probe 1",
+        identifier="28948291",
+        probe_model=pm,
+    )
+    nwbfile.add_device(probe)
+
+    pi = ProbeInsertion(
+        position_reference="(AP, ML, DV) = (0, 0, 0) corresponds to bregma at the cortical surface.",
+        hemisphere="left",
+        depth_in_mm=10.0,
+        insertion_position_ap_in_mm=2.0,
+        insertion_position_ml_in_mm=-4.0,
+        insertion_angle_roll_in_deg=-10.0,
+        insertion_angle_pitch_in_deg=0.0,
+        insertion_angle_yaw_in_deg=0.0,
+    )
+
+    channels_table = ChannelsTable(
+        name="Neuropixels1ChannelsTable",  # test custom name
+        description="Test channels table",
+        reference_mode="Referenced to channel 2.",
+        position_reference="(AP, ML, DV) = (0, 0, 0) corresponds to bregma at the cortical surface.",
+        position_confirmation_method="Histology",
+        probe=probe,
+        probe_insertion=pi,
+        target_tables={
+            "contact": probe.probe_model.contacts_table,
+            "reference_contact": probe.probe_model.contacts_table,
+        },
```

Comment:
Is this a validation issue?

Reply:
Good point. Since `probe`, `probe.probe_model`, and `probe_model.contacts_table` are all required, we can do this. I'll update the PR.